### DISCUSSION TITLE
Update Craft CMS server requirements to more recent versions that are not end of life

### DISCUSSION THREAD

**Discussion by ldennison-bsd** [craftcms/cms#8793] [Ideas]
Currently, the Craft [server requirements page](https://docs.craftcms.com/v3/requirements.html) lists PHP 7.0, MySQL 5.5, and MariaDB 5.5 as minimum versions. These need to be updated to more recent versions that are not end of life (EOL).

Specifically, Craft should require:
- PHP 7.2+
- MySQL 5.6+
- MariaDB 10.1+

PHP 7.0 went EOL in Dec 2018. It's successor PHP 7.1 is also EOL as of Dec 2019. PHP 7.2 is currently the lowest version still in active support mode, but even that is ending in Nov 2020.

MySQL 5.5 went EOL in Dec 2018. Documentation for 5.5 is not even officially available on mysql.com anymore. MySQL 5.6 is the lowest supported version until Feb 2021.

MariaDB 5.5 is still supported, but only until next month (April 2020). MariaDB 10.1 is supported, but even that is ending in Oct 2020.

Generally speaking, in my opinion, Craft CMS should strive to only support versions of underlying software that are actively supported. To support EOL versions of software promotes slack infrastructure security practices, and in the case of PHP especially, poor software engineering practices by building on versions that lack modern features, security improvements, and performance benefits.

**Comment by brandonkelly** [comment]
We choose new system requirements alongside new major releases (so Craft 4.0). Going forward we plan to start releasing new major versions annually, so we shouldn’t get in a situation where we’re still supporting  EOL’d versions of PHP anymore.

**Comment by RitterKnightCreative** [comment]
I wholeheartedly agree with this. 

However, it's also important to look at what versions upstream Linux providers are "actively supporting" with their OS versions as well to get a clear picture.

Just because PHP 7.0 is dead on the OEM side doesn't mean Canonical (Ubuntu)/Debian or RedHat (CentOS) aren't still providing security backports for older software versions that a lot of people still rely on. 

For example, Ubuntu 16.04 is still supported until April 2021 and Debian 9 until 2022. 

I'm not saying it's a great idea to keep running ancient OEM PHP or MySQL versions but it's not really a security problem people make it out to be if you're on an actively supported LTS distro. (Most website compromises are from vulnerable developer code, not core PHP vulnerabilities).

Obviously as developers, it's ideal if people are on the latest versions though and I think PHP's 3 year lookback window is a good place to start. 

FWIW I would fully expect Craft 4 to require PHP 7.4...  or even 8 depending on when it ships. But I think most users would agree having to make sure their PHP version is upgraded every year can be tedious.

**Comment by brandonkelly** [comment]
@RitterKnightCreative We generally will go with whatever the _latest_ Ubuntu LTS release comes with.
